libs/vkd3d/device.c

```diff
@@ -1843,6 +1847,9 @@ static HRESULT vkd3d_select_queues(const struct vkd3d_instance *vkd3d_instance,
     info->family_index[VKD3D_QUEUE_FAMILY_SPARSE_BINDING] = vkd3d_find_queue(count, queue_properties,
             VK_QUEUE_SPARSE_BINDING_BIT, VK_QUEUE_SPARSE_BINDING_BIT);
 
+    /* TODO make this a low-priority compute queue that does not conflict with the actual compute queue */
+    info->family_index[VKD3D_QUEUE_FAMILY_INTERNAL_COMPUTE] = info->family_index[VKD3D_QUEUE_FAMILY_COMPUTE];
+
```

Comment:
Do we intend to add
```

    if (info->family_index[VKD3D_QUEUE_FAMILY_COMPUTE] == VK_QUEUE_FAMILY_IGNORED || VKD3D_FORCE_SINGLE_QUEUE)
        info->family_index[VKD3D_QUEUE_FAMILY_COMPUTE] = info->family_index[VKD3D_QUEUE_FAMILY_GRAPHICS];
```
below?

Reply:
I guess we *always* want to use async compute queue here, but we might have to handle the VK_QUEUE_FAMILY_IGNORED case at least?

Reply:
We actually do need to handle the case where we don't have a compute queue, but not sure if we want to enforce single-queue mode here. Do you have a test case to check whether having a compute queue still leads to bad perf on RADV?

I've tested this in Control, but since the game doesn't constantly allocate and deallocate memory, it wouldn't trigger the issue.

Reply:
I think we want to always use async compute if we can here. I have a test case in Granite still. Can test it next time I swap out to AMD GPU.

Reply:
I could test this on my end if you can share the Granite test.